sqlx-core/src/arguments.rs

```diff
@@ -149,13 +149,36 @@ impl<'a, Db: Database> Arguments<'a, Db> {
         self.positional.iter()
     }
 
+    /// Returns an iterator of the named parameters.
+    pub fn named(&self) -> impl Iterator<Item = &Argument<'a, Db>> {
+        self.named.iter()
+    }
+
     /// Returns a reference to the argument at the location, if present.
     pub fn get<'x, I: ArgumentIndex<'a, Db>>(&'x self, index: &I) -> Option<&'x Argument<'a, Db>> {
         index.get(self)
     }
 }
 
 impl<'a, Db: Database> Argument<'a, Db> {
+    /// Gets the name of this argument, if it is a named argument, None otherwise
+    pub fn name<'b>(&'b self) -> Option<&'a str>{
+        if let Either::Right(name) = self.parameter{
+            Some(name)
+        }else{
+            None
+        }
+    }
+
+    /// Gets the name of this argument, if it is a named argument, None otherwise
```

Comment:
The rest of the API refers to these as positional, can we rename this to `fn position` ? The doc is a copy paste and `rustfmt` needs to be run. Otherwise :+1: